### PR TITLE
Bug 1630714 - No perfherder data displayed for some jobs

### PR DESCRIPTION
**Pull request by airimovici**
Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=1630714
Avoid integrity errors for same tests on different applications

### CODE REVIEW

**Comment by airimovici** [comment]
Forgot to add the migration script

File: tests/model/test_performance_signature.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from datetime import datetime
+
+from treeherder.perf.models import PerformanceSignature
+
+
+def test_performance_signatures_with_different_applications(test_perf_signature):
+    assert PerformanceSignature.objects.count() == 1
+
+    # create a performance signature that only differs from another existing one by the application name
+    PerformanceSignature.objects.create(
```
Comment (ionutgoldan) [inline tests/model/test_performance_signature.py:10]: This is pretty hard to follow.
How about doing a simple clone, similar to [Django's official recommendations](https://docs.djangoproject.com/en/3.0/topics/db/queries/#copying-model-instances)?
What's left for us is to simply change the application attribute, then save the new instance.

**Comment by airimovici** [comment]
The problem here is when we try to ingest results for tests that only differ by application (they have the same suite, extraOptions, framework, repo ...). It fails with an IntegrityError when trying to [create a performance signature](https://github.com/mozilla/treeherder/blob/master/treeherder/etl/perf.py#L44). So, if we already ingested data for a test like `amazon-cold` on `geckoview`-`mozilla-central`-`browsertime`... we'd have a problem ingesting data for the same test on `fenix`.

To solve this we added the `application` in the unique constraint and now the performance_signatures can be created (there is a new unit test that tests the creation of 2 performance signatures that differ only by application).

To test this fix with real ingestion data, we'd need to test it on prototype. But there might be a problem after testing this (and ingesting data for several applications) to revert the database to the previous state (it will have corrupt data, coming back to the old unique constraint). @camd  Is there a good way to revert the db to the previous state after testing this on prototype?

**Comment by camd** [comment]
> The problem here is when we try to ingest results for tests that only differ by application (they have the same suite, extraOptions, framework, repo ...). It fails with an IntegrityError when trying to [create a performance signature](https://github.com/mozilla/treeherder/blob/master/treeherder/etl/perf.py#L44). So, if we already ingested data for a test like `amazon-cold` on `geckoview`-`mozilla-central`-`browsertime`... we'd have a problem ingesting data for the same test on `fenix`.
> 
> To solve this we added the `application` in the unique constraint and now the performance_signatures can be created (there is a new unit test that tests the creation of 2 performance signatures that differ only by application).
> 
> To test this fix with real ingestion data, we'd need to test it on prototype. But there might be a problem after testing this (and ingesting data for several applications) to revert the database to the previous state (it will have corrupt data, coming back to the old unique constraint). @camd Is there a good way to revert the db to the previous state after testing this on prototype?

Prototype data isn't all that important.  If it's a little off, it's no big deal.  We can't really roll-back changes because then you'd lose the latest ingested tasks.  So the best way to get to a previous state would be to duplicate the DB from production or staging over to prototype.  That being said, I don't think we should worry about it.  :)

**Comment by ionutgoldan** [comment]
We did the manual deploy on treeherder-prototype & we managed to confirm the fix.
Merging this to master.
